### PR TITLE
[sync ent] increase segment max limit to 4*64, make configurable (#1424)

### PR DESCRIPTION
**Pull request by FFMMM**
Synching ent-->oss

Author: FFMMM <FFMMM@users.noreply.github.com>
Date:   Thu Dec 9 13:53:44 2021 -0800

    increase segment max limit to 4*64, make configurable (#1424)

    Signed-off-by: FFMMM <FFMMM@users.noreply.github.com>

### CODE REVIEW

File: .changelog/1424.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:feature
```
Comment (rboyer) [inline .changelog/1424.txt:1]: Noting that this file needs to be renamed to `_1424.txt`

**Review by rboyer — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/523311.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/530230.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 74eb257b1c5da75360c5e6aa4c2897e2afd6b46f onto `release/1.11.x` succeeded!

**Pull request merged**